.github/workflows/quality.yml

```
@@ -0,0 +1,95 @@
+    branches: ["main", "master", "dev", "staging"]
+  workflow_dispatch:  # Allow manual trigger
+
+jobs:
+  test-and-lint:
+    runs-on: ubuntu-latest
+    
+    steps:
+    - uses: actions/checkout@v3
+    
+    - name: Set up Python
+      uses: actions/setup-python@v4
```

actionlint (1.7.4)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)